translator/source_code/cotrip_translator.py

```diff
@@ -0,0 +1,421 @@
+
+        for i in range(road_event_length):
+            feature = message['features'][i]
+            road_event_id = road_event_ids[i]
+            feature['properties']['road_event_id'] = road_event_id
+            feature['properties']['data_source_id'] = data_source_id
+            feature['properties']['relationship']['relationship_id'] = str(uuid.uuid4())
+            feature['properties']['relationship']['road_event_id'] = road_event_id 
+
+    return message
+
+
+help_string = """ 
+
+Usage: python icone_translator.py [arguments]
+
+Global options:
+-h, --help                  Print this usage information.
+-i, --input                 specify the xml file to translate
+-o, --output                specify the output file for generated wzdx geojson message """
+
+def parse_arguments(argv):
+    inputfile = ''
+    outputfile = 'cotrip_wzdx_translated_output_message.geojson'
+    try:
+        opts, args = getopt.getopt(argv, "hi:o:", ["input=", "output="])
+    except getopt.GetoptError:
+        sys.exit(2)
+    for opt, arg in opts:
+        if opt in ("-h", "--help"):
+            print(help_string)
+            sys.exit()
+        elif opt in ("-i", "--input"):
+            inputfile = arg
+        elif opt in ("-o", "--output"):
+            outputfile = arg
+    return inputfile, outputfile
+inputfile, outputfile = parse_arguments(sys.argv[1:])
+
+
+def initialize_info():
+    info = {}
+
+    #### Consider whether this id needs to be hardcoded or generated
+    info['feed_info_id'] = "104d7746-688c-44ed-b195-2ee948bf9dfa"
+
+    #### This information is required, might want to hardcode
+    info['metadata'] = {}
+    info['metadata']['wz_location_method'] = "channel-device-method"
+    info['metadata']['lrs_type'] = "lrs_type"
+    info['metadata']['contact_name'] = "Abinash Konersman"  # we can consider to add a representive name from iCone
+    info['metadata']['contact_email'] = "abinash.konersman@state.co.us"
+    info['metadata']['issuing_organization'] = "COtrip"
+
+    return info
+
+
+def parse_xml(inputfile):
+    with open(inputfile, encoding='utf-8-sig') as ficone:
+        # Read
+        xml_string = ficone.read()
+        icone_obj = xmltodict.parse(xml_string)
+        return icone_obj
+
+def validate_wzdx(wzdx_obj, wzdx_schema):
+    #wzdx_schema = json.loads(open(location_schema).read())
+    try:
+      validate(instance=wzdx_obj, schema=wzdx_schema)
+    except ValidationError as e:
+      logging.error(RuntimeError(str(e)))
+      return False
+    return True
+
+def validate_write(wzdx_obj, outputfile, location_schema):
+    wzdx_schema = json.loads(open(location_schema).read())
+    if not validate_wzdx(wzdx_obj, wzdx_schema):
+        return False
+    with open(outputfile, 'w') as fwzdx:
+        fwzdx.write(json.dumps(wzdx_obj, indent=2))
+    return True
+
+
+if inputfile:
+    # Added encoding argument because of weird character at start of alerts.xml file
+
+    icone_obj = parse_xml(inputfile)
```

Comment:
This isn't an icone object, it's an XML object parsed from from cotrip at this point. Let's rename accordingly.